var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -14,6 +14,8 @@ class Hypre(Package):
 
     # hypre does not know how to build shared libraries on Darwin
     variant('shared', default=sys.platform!='darwin', description="Build shared library version (disables static library)")
+    # SuperluDist have conflicting headers with those in Hypre
+    variant('internal-superlu', default=True, description="Use internal Superlu routines")
```

Comment:
I think the sense of this should be inverted.  `+superlu` should build with an external `superlu`, to mirror what other packages do, and you can make it off by default.  If it's off, petsc should use the internal `superlu` routines.